dbms/src/Parsers/ASTCreateQuery.cpp

```diff
@@ -231,7 +235,7 @@ void ASTCreateQuery::formatQueryImpl(const FormatSettings & settings, FormatStat
             << (!as_database.empty() ? backQuoteIfNeed(as_database) + "." : "") << backQuoteIfNeed(as_table);
     }
 
-    if (columns_list)
+    if (columns_list && !as_table_function)
```

Comment:
It shouldn't be able to parse CREATE table (columns_list) AS table_function(...)
Please add negative test.

Reply:
Test will be added, but this code actually prevents printing a frame with columns in metadata sql file. so it can simply be ATTACH table AS table_function(). Here comes the next question, why was columns_list initialized. (It responds to your last comment on this topic) It is initialized with setColumns()  to avoid duplicating some code in initialization of a table. But I am not sure if is right.

Reply:
> It shouldn't be able to parse CREATE table (columns_list) AS table_function(...)
> Please add negative test.

Test will be added, but this code actually prevents printing a frame with columns in metadata sql file. so it can simply be ATTACH table AS table_function(). Here comes the next question, why was columns_list initialized. (It responds to your last comment on this topic) It is initialized with setColumns() to avoid duplicating some code in initialization of a table. But I am not sure if is right.